system/mp/mpcomm.cpp

```diff
@@ -3023,6 +3023,8 @@ class CCommunicator: public ICommunicator, public CInterface
                     unsigned remaining;
                     if (tm.timedout(&remaining))
                         return false;
+                    if (skipDisconnected && !channel->isConnected())
```

Comment:
not that keen on handling the disconnects conditionally at this level.
I think it would be better to conditionally send based on whether it's connected (see comment in broadcast)